### PR TITLE
Agenda sequence dedupe

### PR DESCRIPTION
**Pull request by reginafcompton**
This PR addresses https://github.com/datamade/la-metro-councilmatic/issues/337.

As the issue details, Legistar has – on at least one known occasion – duplicated the 'EventItemAgendaSequence'. This PR handles the problem by checking if an agenda has duplicate 'EventItemAgendaSequence', and if so, raising a ValueError with an instructive message to contact Metro. 

N.b., later on, we may implement a solution that flags problematic Events and, rather than raising an error, renders a message to Metro admin further downstream.

### CODE REVIEW

File: lametro/events.py [hunk 1]
```diff
@@ -206,6 +205,17 @@ def scrape(self, window=None) :
...
+                    # The EventItemAgendaSequence provides 
+                    # the line number of the Legistar agenda grid.
+                    agenda_item['extras']['item_agenda_sequence'] = item['EventItemAgendaSequence']
+
+                # Historically, the Legistar system has duplicated the EventItemAgendaSequence,
+                # resulting in data inaccuracies. The scrape should fail in such cases, until Metro
+                # cleans the data.
+                item_agenda_sequences = [item['extras']['item_agenda_sequence'] for item in e.agenda]
+                if len(item_agenda_sequences) != len(set(item_agenda_sequences)):
+                    raise ValueError('Agenda for Event {} has duplicate agenda items on the Legistar grid. Contact Metro, and ask them to remove the duplicate EventItemAgendaSequence.'.format(e.pupa_id))
```
Comment (hancush) [inline lametro/events.py]: could we include the identifier (or something that would allow metro to locate the duplicates) in this message? that way, we can act on them without having to do any further sleuthing.

File: tests/lametro/test_events.py [hunk 2]
```diff
@@ -30,4 +30,28 @@ def test_status_assignment(event_scraper,
...
+        m.get(matcher, json={}, status_code=200)
+
+        matcher = re.compile('metro.legistar.com')
+        m.get(matcher, json={}, status_code=200)
+
+        api_event['event_details'] = {'note': 'web', 
+                                      'url': 'https://metro.legistar.com/MeetingDetail.aspx?ID=642118&GUID=F19B2133-928C-4390-9566-C293C61DC89A&Options=info&Search='}
+
+        mocker.patch('lametro.LametroEventScraper._merge_events', return_value=[(api_event, web_event)])
+        mocker.patch('lametro.LametroEventScraper.agenda', return_value=[event_agenda_item, event_agenda_item])
```
Comment (hancush) [inline tests/lametro/test_events.py]: nice. :-) actual question: should we also test that no duplicates are scraped without error? (or is this a given?)

File: lametro/events.py [hunk 3]
```diff
@@ -206,6 +205,17 @@ def scrape(self, window=None) :
...
 
+                    # The EventItemAgendaSequence provides 
+                    # the line number of the Legistar agenda grid.
+                    agenda_item['extras']['item_agenda_sequence'] = item['EventItemAgendaSequence']
+
+                # Historically, the Legistar system has duplicated the EventItemAgendaSequence,
+                # resulting in data inaccuracies. The scrape should fail in such cases, until Metro
+                # cleans the data.
+                item_agenda_sequences = [item['extras']['item_agenda_sequence'] for item in e.agenda]
+                if len(item_agenda_sequences) != len(set(item_agenda_sequences)):
```
Comment (hancush) [inline lametro/events.py]: clever! :-)

File: tests/lametro/test_events.py (hunk 2 above)
Comment (reginafcompton) [reply tests/lametro/test_events.py]: Oh, yes: we should. I got lazy.

Comment (hancush) [reply tests/lametro/test_events.py]: you: lazy once. me: proactively lazy, potentially many times. ha.

File: lametro/events.py (hunk 1 above)
Comment (reginafcompton) [reply lametro/events.py]: yup, yup, yup

File: lametro/events.py [hunk 4]
```diff
@@ -188,6 +188,9 @@ def scrape(self, window=None) :
             # it exists, to the extras dict.
             e.extras = {'guid': event['EventGuid']}
 
+            legistar_api_url = self.BASE_URL + '/events/{0}'.format(event['EventId'])
+            e.add_source(legistar_api_url, note='api')
+
```
Comment (reginafcompton) [inline lametro/events.py]: I moved this up, so that I could easily create `legistar_api_url` and reuse it on line 221.
